### ISSUE TITLE
Compress option not working in CLI - possible conflict with other options

### ISSUE THREAD

**Issue by dhruvbhatia** [cssnext/cssnext#195]
Using the `-c` flag within the CLI does not actually compress a given input file under `v1.8.3`.

E.g. let's consider the [compression example](http://cssnano.co/) on the cssnano homepage:

**input.css**
```css
h1::before, h1:before {
    margin: 10px 20px 10px 20px;
    color: #ff0000;
    -webkit-border-radius: 16px;
    border-radius: 16px;
    font-weight: normal;
    font-weight: normal;
}
/* invalid placement */
@charset "utf-8";
```

We'd expect `cssnext -c input.css` to give us this:
```css
@charset "utf-8";h1:before{margin:10px 20px;color:red;border-radius:1pc;font-weight:400}
```

However we just get back the original `input.css` (non-compressed).

After some digging I found that adding the following flags in addition to `-c` seems to help, so I imagine there's a conflict with how cssnano is being called:

**Hacky workaround - disable everything but compression:**
`> cssnext -c input.css --no-custom-properties --no-calc --no-custom-media --no-media-queries-range --no-custom-selectors --no-color-rebeccapurple --no-color-hwb --no-color-gray --no-color-hex-alpha --no-color-function --no-font-variant --no-filter --no-rem --no-pseudo-elements --no-pseudo-class-matches --no-pseudo-class-not --no-pseudo-class-any-link --no-color-rgba --no-autoprefixer`

The above delivers compressed output, but it obviously isn't ideal given we're effectively disabling all the other features within cssnext.

Has anyone run into the same issue?

**Comment by ben-eb** [comment]
Might be related to this: https://github.com/cssnext/cssnext/issues/194

**Comment by MoOx** [comment]
Not sure how it's related since nobody raised this issue until now.
@dhruvbhatia do you get the same issue without the charset rule?

**Comment by dhruvbhatia** [comment]
Thanks for the quick responses!

Yes I get the same issue, even without the charset rule. I.e. Even basic input isn't compressed.

If you can't reproduce then perhaps there's something wrong at my end? Are you able to produce the expected output in the example above?

**Comment by dhruvbhatia** [comment]
I tested historical versions in clean environments and discovered that the unexpected behaviour only started in `v1.8.0` (compression worked fine prior).

This coincides with the upgrade to cssnano `v2.x` - #166 as per the [v1.8.0 release notes](https://github.com/cssnext/cssnext/blob/master/CHANGELOG.md#180---2015-06-29).

`v1.7.1` works fine so using that for the time being, but I'll keep digging and see if I can isolate further..

**Comment by ymauray** [comment]
I confirm that starting from v1.8.0, compression does not work. I use cssnext as a PostCSS plugin.

**Comment by apapirovski** [comment]
@ben-eb seems correct to me based on my testing. When I changed postcssPlugin names to be unique, everything started working again.

**Comment by MoOx** [comment]
Tests were passing when 1.8.3 were released, and now the same tests are broken with a fresh install.
So this is related to a breaking change introduced by a recent cssnano "minor" update (that should have been major ?).
I will fix that by addressing #194

**Comment by ben-eb** [comment]
You can work around this by passing this config to cssnext:

```js
{
  compress: {filterPlugins: false}
}
```

**Comment by MoOx** [comment]
Fix released as 1.8.4
